plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/N4JSScopeProvider.xtend

```diff
@@ -533,4 +565,58 @@ class N4JSScopeProvider extends AbstractScopeProvider implements IDelegatingScop
 		if (obj === null) return null;
 		return EcoreUtil2.getContainerOfType(obj.eContainer, ancestorType);
 	}
+	
+	
+	private def getN4JSXScope(EObject context, EReference reference) {
+		val jsxPropertyAttributeScope = getJSXPropertyAttributeScope(context, reference)
+		if(jsxPropertyAttributeScope !== IScope.NULLSCOPE)
+			return jsxPropertyAttributeScope
+
+		val jsxElementScope = getJSXElementScope(context, reference)
+		if(jsxElementScope !== IScope.NULLSCOPE)
+			return jsxElementScope
+
+		//delegate to host -> N4JS scope
+		return getN4JSScope(context, reference);
+	}
+	/** Returns scope for the {@link JSXPropertyAttribute} (obtained from context) or {@link IScope#NULLSCOPE} */
+	private def getJSXPropertyAttributeScope(EObject context, EReference reference) {
+		if (reference == N4JSPackage.Literals.JSX_PROPERTY_ATTRIBUTE__PROPERTY) {
+			if (context instanceof JSXPropertyAttribute) {
+				val jsxElem = (context.eContainer as JSXElement);
+				val TypeRef propsTypeRef = jsxElem.getPropsType();
+				val checkVisibility = true;
+				val staticAccess = false;
+				if (propsTypeRef !== null) {
+					// Prevent "Cannot resolve to element" error message of unknown attributes since
+					// we want to issue a warning instead
+					val memberScope = memberScopingHelper.createMemberScope(propsTypeRef, context, checkVisibility,
+						staticAccess);
+					return new DynamicPseudoScope(memberScope);
+				} else {
+					val scope = getN4JSScope(context, reference);
+					return new DynamicPseudoScope(scope);
+				}
+			}
+		}
+		return IScope.NULLSCOPE;
+	}
+	
+	/** Returns scope for the JSXElement (obtained from context) or {@link IScope#NULLSCOPE} */
+	private def getJSXElementScope(EObject context, EReference reference) {
+
+		// Otherwise, if we are within a JSX element context but cannot bind JSX element, ignore binding for now
+		var JSXElementName jsxElName;
+		for (var ei = context; ei !== null; ei = ei.eContainer) {
+			if (ei instanceof JSXElementName) {
+				jsxElName = ei;
+			}
+		}
```

Comment:
Not your code, I know, but please replace this loop by `EcoreUtil2.getContainerOfType(EObject, Class<T>)`, if possible.

Reply:
changed